tests/unit_tests/rigid_pendulum_poc/test_generalized_alpha_solver.cpp

```diff
@@ -0,0 +1,246 @@
+    auto residual_force = create_vector({1.e-7, 2.e-7, 3.e-7});
+    auto incremental_force = create_vector({1., 2., 3.});
+    auto time_integrator = GeneralizedAlphaTimeIntegrator(0., 1., 1);
+    auto converged = time_integrator.CheckConvergence(residual_force, incremental_force);
+
+    EXPECT_TRUE(converged);
+}
+
+TEST(TimeIntegratorTest, ExpectNonConvergedSolution) {
+    auto residual_force = create_vector({1.e-3, 2.e-3, 3.e-3});
+    auto incremental_force = create_vector({1., 2., 3.});
+    auto time_integrator = GeneralizedAlphaTimeIntegrator(0., 1., 1);
+    auto converged = time_integrator.CheckConvergence(residual_force, incremental_force);
+
+    EXPECT_FALSE(converged);
+}
+
+TEST(GeneralizedAlphaTimeIntegratorTest, ConstructorWithInvalidAlphaF) {
+    EXPECT_THROW(
+        GeneralizedAlphaTimeIntegrator(0., 1., 1, 1.1, 0.5, 0.25, 0.5, 10), std::invalid_argument
+    );
+}
+
+TEST(GeneralizedAlphaTimeIntegratorTest, ConstructorWithInvalidAlphaM) {
+    EXPECT_THROW(
+        GeneralizedAlphaTimeIntegrator(0., 1., 1, 0.5, 1.1, 0.25, 0.5, 10), std::invalid_argument
+    );
+}
+
+TEST(GeneralizedAlphaTimeIntegratorTest, ConstructorWithInvalidBeta) {
+    EXPECT_THROW(
+        GeneralizedAlphaTimeIntegrator(0., 1., 1, 0.5, 0.5, 0.75, 0.5, 10), std::invalid_argument
+    );
+}
+
+TEST(GeneralizedAlphaTimeIntegratorTest, ConstructorWithInvalidGamma) {
+    EXPECT_THROW(
+        GeneralizedAlphaTimeIntegrator(0., 1., 1, 0.5, 0.5, 0.25, 1.1, 10), std::invalid_argument
+    );
+}
+
+TEST(GeneralizedAlphaTimeIntegratorTest, ConstructorWithInvalidNumberOfSteps) {
+    EXPECT_THROW(
+        GeneralizedAlphaTimeIntegrator(0., 1., 1, 0.5, 0.5, 0.25, 0.5, 0), std::invalid_argument
+    );
+}
+
+TEST(GeneralizedAlphaTimeIntegratorTest, GetDefaultGAConstants) {
+    auto time_integrator = GeneralizedAlphaTimeIntegrator();
+
+    EXPECT_EQ(time_integrator.GetAlphaF(), 0.5);
+    EXPECT_EQ(time_integrator.GetAlphaM(), 0.5);
+    EXPECT_EQ(time_integrator.GetBeta(), 0.25);
+    EXPECT_EQ(time_integrator.GetGamma(), 0.5);
+    EXPECT_EQ(time_integrator.GetMaxIterations(), 10);
+}
+
+TEST(GeneralizedAlphaTimeIntegratorTest, GetSuppliedGAConstants) {
+    auto time_integrator = GeneralizedAlphaTimeIntegrator(0., 1., 1, 0.11, 0.29, 0.47, 0.93, 17);
+
+    EXPECT_EQ(time_integrator.GetAlphaF(), 0.11);
+    EXPECT_EQ(time_integrator.GetAlphaM(), 0.29);
+    EXPECT_EQ(time_integrator.GetBeta(), 0.47);
+    EXPECT_EQ(time_integrator.GetGamma(), 0.93);
+    EXPECT_EQ(time_integrator.GetMaxIterations(), 17);
+}
+
+TEST(TimeIntegratorTest, AlphaStepSolutionAfterOneIncWithZeroAcceleration) {
+    auto initial_state = State();
+
+    expect_kokkos_view_1D_equal(initial_state.GetGeneralizedCoordinates(), {0.});
+    expect_kokkos_view_1D_equal(initial_state.GetGeneralizedVelocity(), {0.});
+    expect_kokkos_view_1D_equal(initial_state.GetGeneralizedAcceleration(), {0.});
+    expect_kokkos_view_1D_equal(initial_state.GetAlgorithmicAcceleration(), {0.});
```

Comment:
This is tested elsewhere, so there's no need to put these checks here.  Removing unnecessary checks makes your tests shorter and easier to read, plus it ensures that you can easily see what new features are exercised by this test.